libs/messaging.ts

```diff
@@ -0,0 +1,6 @@
+import { defineExtensionMessaging } from "@webext-core/messaging";
+
+type SearchMessaging = {
+  searchOnTab: (data: { url: string }) => void;
+};
+export const searchMessaging = defineExtensionMessaging<SearchMessaging>();
```

Comment:
wow, its cool that it infer event types.

Reply:
Aaron (the author of wxt) is making this one too lol.